src/EditorFeatures/Core.Wpf/InlineRename/UI/Adornment/RenameFlyout.xaml.cs

```diff
@@ -221,56 +237,59 @@ private void Adornment_ConsumeMouseEvent(object sender, MouseButtonEventArgs e)
 
         private void Adornment_GotKeyboardFocus(object sender, KeyboardFocusChangedEventArgs e)
         {
-            if (e.OldFocus == this)
+            if (e.NewFocus != RenameUserInput)
             {
-                return;
+                RenameUserInput.Focus();
+                e.Handled = true;
             }
-
-            IdentifierTextBox.Focus();
-            e.Handled = true;
         }
 
         private void ToggleExpand(object sender, RoutedEventArgs e)
         {
             _viewModel.IsExpanded = !_viewModel.IsExpanded;
         }
 
+        private void RenameUserInput_GotFocus(object sender, RoutedEventArgs e)
+        {
+            this.RenameUserInput.SelectAllText();
```

Comment:
This feels like an implementation detail of `IRenameUserInput`, is there a reason the flyout handles this?

Reply:
Mostly to centralize the text selection logic, so we don't have to spread "where do we select the text" in every implementation of `IRenameUserInput`.